pkg/sql/opt/ordering/lookup_join.go

```diff
@@ -82,7 +82,8 @@ func indexJoinBuildProvided(expr memo.RelExpr, required *props.OrderingChoice) o
 	// using column equivalencies.
 	indexJoin := expr.(*memo.IndexJoinExpr)
 	rel := indexJoin.Relational()
-	return remapProvided(indexJoin.Input.ProvidedPhysical().Ordering, &rel.FuncDeps, rel.OutputCols)
+	input := indexJoin.Input
+	return remapProvided(input.ProvidedPhysical().Ordering, &input.Relational().FuncDeps, rel.OutputCols)
```

Comment:
Are there downsides to using the input's FDs? It seems like we'd potentially break other cases that rely on the FD containing columns that are fetched by the index join.